### DISCUSSION TITLE
Lazy commands

### DISCUSSION THREAD

**Discussion by Yohe-Am** [c4spar/cliffy#696] [Q&A]
Hey, thanks you for the good software before anything.

Som I'm writing this app and some sections of the CLI are dynamically generated from configuration. The configuration is in typescript and thus it's required to go through a worker to get to it. I have caching in place to avoid this expensive call but I'd still like to eliminate to the cost when the user access sections of the CLI that don't require this. 

Is it possible to implement something like like this?

**Comment by c4spar** [comment]
I had plans to add lazy loading for commands but haven't had time to do it yet.

Something like this:

```ts
new Command()
  .command("foo", () => import("./foo-command.ts").then(({fooCommand)}) => fooCommand)
```

**Reply by Yohe-Am** [reply]
Should I open a ticket ?

**Reply by c4spar** [reply]
Feel free to open a new issue 👍
